packages/shared/src/components/fields/Switch.tsx

```diff
@@ -41,6 +41,7 @@ function SwitchComponent(
         className,
         'group relative flex items-center',
         disabled ? 'pointer-events-none cursor-not-allowed' : 'cursor-pointer',
+        disabled && checked && 'opacity-16',
```

Comment:
Currently when the switch is disabled and checked, there isn't any darkening effect, so we add it here.